spec/wings/services/custom_queries/find_collections_by_type_spec.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+require 'wings_helper'
+require 'wings/services/custom_queries/find_collections_by_type'
+
+RSpec.describe Wings::CustomQueries::FindCollectionsByType, :clean_repo do
+  subject(:query_handler) { described_class.new(query_service: query_service) }
+  let(:collection_type)   { FactoryBot.create(:collection_type) }
+  let(:type_gid)          { collection_type.to_global_id }
+  let(:query_service)     { Hyrax.query_service }
+
+  describe '#find_collections_by_type' do
+    context 'when there are no collections' do
+      it 'is empty' do
+        expect(query_handler.find_collections_by_type(global_id: type_gid))
+          .to be_empty
+      end
+    end
+
+    context 'when collections exist' do
+      let(:collection_with_default_type) { FactoryBot.create(:collection) }
+      let(:non_collection_model)         { FactoryBot.create(:work) }
+
+      let(:collections_with_target_type) do
+        FactoryBot.create_list(:collection_lw, 3, collection_type_gid: type_gid)
+      end
+
+      before do # force create all the models
+        non_collection_model
+        collection_with_default_type
+        collections_with_target_type
+      end
```

Comment:
I'm curious of your thoughts on lazy loading these in a before block vs. using `let!`.

Reply:
i think rubocop bars `let!` and this is my attempt at a slightly more intent-revealing approach for that.

Reply:
Interesting.  I tend to think that `let!` is clear (except that the little `!` can be easily overlooked) and I usually find the simple references in a before block odd-looking but your comment makes it better by making the intent explicit.

Reply:
the issue with `let!`, i think, is that it leads folks to write test suites that eagerly create unneeded test objects. my understanding is that the style restriction seeks to make this deliberately more cumbersome/visible.